content/docs/1.0.0/tutorial/create-your-first-microservice-javascript.md

```diff
@@ -76,97 +88,91 @@ const iofog = require('@iofog/nodejs-sdk');
 
-Because a config can be any arbitrary JSON, including very large files, the change notifications themselves do not actually include the config. So if you do in fact want to update your local cache of the config, you can follow up a change notification with a call to `iofog.getConfig()`.
+Therefore we have to connect the ioFog *control channel* via WebSocket, which is used to receive notifications from the Controller that our microservice's config has changed.
+
+Because a config can be any arbitrary JSON, including very large files, the change notifications themselves do not actually include the config. So if you do in fact want to update your local cache of the config, you have to follow up a change notification with a call to `iofog.getConfig()`.
 
 ```js
 iofog.wsControlConnection({
-  onNewConfigSignal: () => {
-    // notification-only, if you want to the updated config
-    // you'll need to subsequently use iofog.getConfig()
-    console.log('config has changed');
-  },
-  onError: err => console.error('Error with Control Connection: ', err)
+    onNewConfigSignal: () => updateConfig(),
+    onError: err => console.error('Error with Control Connection: ', err)
 });
 ```
 
 #### iofog.wsMessageConnection()
 
-Connect to the ioFog message channel via WebSocket. This is where you'll receive any messages routed to this microservice from another.
+Next, we have to connect to the ioFog *message channel* via WebSocket. This is where you'll receive any messages routed to this microservice from another.
 
 Under the hood, communication is brokered by your [Connector](../connectors/overview.html) and messages are routed according to that microservice's route settings on the Controller.
 
 ```js
 iofog.wsMessageConnection(onMessageConnectionOpen, {
-  onMessages: messages => {
-    // Do something with incoming messages
-    console.log(messages);
-  },
-  onMessageReceipt: (messageId, timestamp) => {
-    console.log('message receipt: ', { messageId, timestamp });
-  },
-  onError: err => console.error('Message WebSocket error: ', err)
+    onMessages: messages => {
+        // Do something with messages... 
+    },
+    onMessageReceipt: (messageId, timestamp) => {
+        console.log('message receipt: ', {
+            messageId,
+            timestamp
+        });
+    },
+    onError: err => console.error('Message WebSocket error: ', err)
 });
 ```
 
 #### iofog.ioMessage() / iofog.wsSendMessage()
 
-Create and send ioMessages in JSON, which is the Node.js serialization format used for intercommunication between microservices.
+Now that we can read control signals and message, we also need to send messages out with the actual moving average. We create and send ioMessages in JSON, which is the Node.js serialization format used for intercommunication between microservices.
 
 When your code wants to publish a message to any other microservice, these are what you'll be sending.
```

Comment:
When **our** code wants to publish a message to any other microservice, these are what **we'll** be sending.